src/Core/Auth/Identity/EmailTokenProvider.cs

```diff
@@ -3,17 +3,30 @@
 using Bit.Core.Entities;
 using Bit.Core.Services;
 using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Caching.Distributed;
 using Microsoft.Extensions.DependencyInjection;
 
 namespace Bit.Core.Auth.Identity;
 
 public class EmailTokenProvider : IUserTwoFactorTokenProvider<User>
 {
+    private const string CacheKeyFormat = "Email_TOTP_{0}_{1}";
+
     private readonly IServiceProvider _serviceProvider;
+    private readonly IDistributedCache _distributedCache;
+    private readonly DistributedCacheEntryOptions _distributedCacheEntryOptions;
 
-    public EmailTokenProvider(IServiceProvider serviceProvider)
+    public EmailTokenProvider(
+        IServiceProvider serviceProvider,
+        [FromKeyedServices("persistent")]
+        IDistributedCache distributedCache)
     {
         _serviceProvider = serviceProvider;
+        _distributedCache = distributedCache;
+        _distributedCacheEntryOptions = new DistributedCacheEntryOptions
+        {
+            AbsoluteExpirationRelativeToNow = TimeSpan.FromMinutes(30)
```

Comment:
❓ Same.

Reply:
I just made it up. 10x the time window of the TOTP code. Email TOTP window is 3 minutes